### PR TITLE
Update basic typing

### PR DESCRIPTION
**Pull request by jrapin**
## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the **CONTRIBUTING** document and completed the CLA (see **CONTRIBUTING**).
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: nevergrad/functions/powersystems/core.py [hunk 1]
```diff
@@ -6,52 +6,50 @@
...
 
-    def GetParamNumbers(self):
+    def GetParamNumbers(self) -> int:
         return sum([np.prod(l.shape) for l in self.layers])
 
-    def SetParams(self, ww):
+    def SetParams(self, ww: Any) -> None:
         w = [w for w in ww]
-        assert(len(w) == self.GetParamNumbers())
+        assert len(w) == self.GetParamNumbers()
```
Comment (jrapin) [inline nevergrad/functions/powersystems/core.py]: @teytaud no parens for `assert` keyword

File: nevergrad/functions/powersystems/core.py [hunk 2]
```diff
@@ -6,52 +6,50 @@
...
         for l in self.layers:
             s = np.prod(l.shape)
-            l = np.reshape(np.array(w[:s]), l.shape)
+            l = np.reshape(np.array(w[:s]), l.shape)  # TODO @oteytaud new name?
```
Comment (jrapin) [inline nevergrad/functions/powersystems/core.py]: @teytaud this probably fails somehow since l is overridden by the for loop, can you check it?

File: nevergrad/functions/powersystems/core.py [hunk 3]
```diff
@@ -6,52 +6,50 @@
...
         self.layers: List[Any] = []
         self.layers += [np.zeros((input_size, layer_width))]
-        for i in range(layers-2):
+        for _ in range(layers - 2):
             self.layers += [np.zeros((layer_width, layer_width))]
         self.layers += [np.zeros((layer_width, output_size))]
         assert len(self.layers) == layers
 
-    def GetParamNumbers(self):
+    def GetParamNumbers(self) -> int:
```
Comment (jrapin) [inline nevergrad/functions/powersystems/core.py]: @teytaud I'll update method names eventually, there should not be caps ;)

File: nevergrad/functions/powersystems/core.py [hunk 4]
```diff
@@ -79,15 +78,17 @@ def __init__(self, num_dams: int = 13, depth: int = 3, width: int = 3,
         self.back_to_normal = back_to_normal
         self.consumption_noise = consumption_noise
         self.num_thermal_plants = num_thermal_plants
-        self.number_of_years = num_years 
+        self.number_of_years = num_years
         self.failure_cost = failure_cost
-        
+        self.hydro_prod_per_time_step: List[Any] = []  # TODO @oteytaud initial values?
```
Comment (jrapin) [inline nevergrad/functions/powersystems/core.py]: Is this a good initial value? There was none initially so the linter complained

File: nevergrad/examples/powersystem.py [hunk 5]
```diff
@@ -23,12 +28,11 @@
 #     num_years: int = 1,  # Number of years.
 #     failure_cost: float = 500.,  # Cost of not satisfying the demand. Equivalent to an expensive infinite capacity thermal plant.
 
-                    power_system_loss = PowerSystem(num_dams=num_dams, depth=depth, width=width, year_to_day_ratio=year_to_day_ratio, back_to_normal=back_to_normal, num_thermal_plants=num_thermal_plants, constant_to_year_ratio=constant_to_year_ratio) 
-                    optimizer = optimizerlib.SplitOptimizer9(instrumentation=power_system_loss.dimension, budget=budget, num_workers=10)
```
Comment (jrapin) [inline nevergrad/examples/powersystem.py]: @teytaud Did you succeed in making this work??? `SplitOptimizer9` does not exist if I'm correct, it's supposed to be `SplitOptimizer`

File: nevergrad/functions/powersystems/core.py (hunk 1 above)
Comment (teytaud) [reply nevergrad/functions/powersystems/core.py]: thx

File: nevergrad/functions/powersystems/core.py (hunk 2 above)
Comment (teytaud) [reply nevergrad/functions/powersystems/core.py]: not sure, but I do a PR at least for readability

Comment (teytaud) [reply nevergrad/functions/powersystems/core.py]: actually I believe this was correct... in particular we get a decent comparison between algorithms in 
https://dl.fbaipublicfiles.com/nevergrad/allxps/list.html (powersystemsbig) which can not be due to random.

that said readability is a good reason

Comment (jrapin) [reply nevergrad/functions/powersystems/core.py]: I don't understand your answer.
My problem here is that this statement is pointless, since the value in l is lost when it is reset by the for loop. So either the line is not useful, or it is incorrect.
